.devcontainer/rust-zen/Dockerfile

```diff
@@ -0,0 +1,121 @@
+    git \
+    graphviz \
+    make \
+    pkg-config \
+    ssh \
+    wget \
+    vim \
+    less \
+    zip \
+    # python
+    python3-dev \
+    && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/*
+
+################################################################################
+# Bazelisk
+################################################################################
+
+ARG BAZELISK_VERSION=v1.10.1
+ARG BAZELISK_DOWNLOAD_SHA=dev-mode
+RUN curl -fSsL -o /usr/local/bin/bazelisk https://github.com/bazelbuild/bazelisk/releases/download/${BAZELISK_VERSION}/bazelisk-linux-amd64  && \
+    ([ "${BAZELISK_DOWNLOAD_SHA}" = "dev-mode" ] || echo "${BAZELISK_DOWNLOAD_SHA} */usr/local/bin/bazelisk" | sha256sum --check - ) && \
+    chmod 0755 /usr/local/bin/bazelisk && \
+    # The only way to get root (CI), vscode (codespaces) & zen
+    # (local devcontainer) on the same page without a
+    # three-way .bazelrc config
+    sudo ln -s /usr/local/bin/bazelisk /usr/local/bin/bazel
+
+ENV USE_BAZEL_VERSION "${BAZEL_VERSION}"
+
+
+################################################################################
+# Login Shell
+################################################################################
+
+ENV TERM xterm-256color
+ENTRYPOINT ["/bin/bash", "--login", "-i"]
+
+################################################################################
+# User 'zen'
+################################################################################
+
+ARG USERNAME=zen
+
+# Automatically creates a UID, group ${USERNAME}, adds ${USERNAME} to the group
+#
+# NB: If using this image as a base for a devcontainer, it is critical to have
+# just one user, since dynamic UID mapping will not work otherwise, refer to:
+#   https://github.com/microsoft/vscode-remote-release/issues/1155
+#
+RUN useradd -o --uid 1000 -s "/bin/bash" -m ${USERNAME} && \
+    apt-get install -y sudo && \
+    echo "${USERNAME} ALL=NOPASSWD: ALL" > /etc/sudoers.d/${USERNAME} && \
+    chmod 0440 /etc/sudoers.d/${USERNAME}
+
+################################################################################
+# Rust toolchain
+################################################################################
+# Doc: https://rust-lang.github.io/rustup/installation/index.html#choosing-where-to-install
+
+USER ${USERNAME}
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y && \
+    . $HOME/.cargo/env && \
+    rustup update && \
+    rustup default ${RUSTC_VERSION}
+USER root
+
+# RUN  export CARGO_HOME=/home/cargo && export RUSTUP_HOME=/home/rustup && \
+#     curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > rustup-init.sh && \
+#     sh rustup-init.sh --default-toolchain stable -y && \
+#     . $CARGO_HOME/env && \
+#     rustup update && \
+#     rustup default ${RUSTC_VERSION}
+# 
+# RUN echo "source /home/cargo/env" >> /home/${USERNAME}/.bashrc
+# RUN echo "export PATH=\$PATH:$CARGO_HOME/bin" >> /home/${USERNAME}/.bashrc
+
```

Comment:
I have an issue here when working with devcontainers.

Apparently, when the devcontainer is built a `vscode` user is used, so the rust toolchain installation goes under `/home/vscode/.cargo` and `/home/vscode/.rustup` .
So when I connect with a remote user (e.g.: as `zen`) the cargo / rust related stuff isn't sourced/installed so I need to run an extra `rosdep update && rosdep default 1.75.0` command, which obviously isn't convenient as I want everything set up when I connect to the devcontainer.

Summoning @stonier as you have way more experience working with devcontainers.
This made me not fully understand the building process with the devcontainer.
Is the image built with a `vscode` username(or vscode's default)? How is it that the exports ("export PS1..") that are located below the code I marked are being echoed to the right `.bashrc` (/home/zen/.bashrc)?

Reply:
From F2F chat: https://gist.github.com/stonier/e9edd298308a2554d447d2a3dcca8c5e